docs/_posts/2017-09-26-react-v16.0.md

```diff
@@ -0,0 +1,133 @@
+
+### New `render` return types: fragments and strings
+
+You can now return an array of elements from a component's `render` method:
+
+```js
+render() {
+  // No need to wrap list items in an extra element!
+  return [
+    // Don't forget the keys :)
+    <li key="A"/>First item</li>,
+    <li key="B"/>Second item</li>,
+    <li key="C"/>Third item</li>,
+  ];
+}
+```
+
+Support for returning strings from the render method has also been added.
+
+[API Documentation](/docs/react-component.html#render)
+
+### Better error handling
+
+
+Previously, runtime errors during rendering could potentially put React in a broken state, producing cryptic error messages and requiring a page refresh to recover. To address this problem, React 16 uses a more resilient error-handling strategy. By default, if an error is thrown inside a component's render or lifecycle methods, the whole component tree is unmounted from the root. This prevents the display of corrupted data. However, it's probably not the ideal user experience.
+
+Instead of unmounting the whole app every time there's an error, you can use error boundaries. Error boundaries are special components that capture errors inside their subtree and display a fallback UI in its place. Think of error boundaries like try-catch statements, but for React components.
+
+For more details, check out our [previous post on error handling in React 16](/blog/2017/07/26/error-handling-in-react-16.html).
+
+
+### Portals
+
+Portals provide a first-class way to render children into a DOM node that exists outside the DOM hierarchy of the parent component.
+
+```js{6}
+render() {
+  // React does *not* create a new div. It renders the children into `divNode`.
+  // `divNode` is any valid DOM node, regardless of its location in the DOM.
+  return React.createPortal(
+    this.props.children,
+    divNode,
+  );
+}
+```
+
+See a full example in the [documentation for portals](/docs/portals.html)
+
+### Better server-side rendering
+
+React 16 is better at hydrating server-rendered HTML. It no longer requires the initial render to exactly match the result from the server. Instead, it will attempt to reuse as much of the existing DOM as possible. No more checksums!
+
+The server renderer has also been completely rewritten to support streaming. React core team member Sasha Aicken, who worked on this feature, wrote a [great article describing React 16's SSR improvements](https://medium.com/@aickin/whats-new-with-server-side-rendering-in-react-16-9b0d78585d67). "Rendering to a stream can reduce the time to first byte (TTFB) for your content, sending the beginning of the document down the wire to the browser before the next part of the document has even been generated. All major browsers will start parsing and rendering the document earlier when the content is streamed from the server this way."
+
```

Comment:
IMO we should also mention performance improvements here, again, maybe quoting Sasha's article.

Reply:
Specifically that you don't have to bundle the server app anymore to get good performance.

Reply:
Yes. Also just that it's faster 'cause we rewrote it. This is more interesting to most people than that it uses a different hydration strategy.